hoomd/hpmc/integrate.py

```diff
@@ -425,9 +423,11 @@ def __init__(self, seed, d=0.1, a=0.1, move_ratio=0.5,
         typeparam_shape = TypeParameter('shape', type_kind='particle_types',
                                         param_dict=TypeParameterDict(
                                             vertices=list,
-                                            ignore_statistics=False,
-                                            sweep_radius=0,
-                                            len_keys=1)
+                                            ignore_statistics=bool,
```

Comment:
`ignore_statistic` and `sweep_radius` were fine. `explicit_defaults` is only for the case where you have a more complex validation function to specify. Otherwise `TypeParameterDict` will be able to tell that `False` is a boolean value and `0` is an integer. If you want a float `0.` will be recognized as such.